### PR TITLE
Fix path generation in IP Adapter

### LINKED ISSUE #6561: OSError: h94/IP-Adapter does not appear to have a file named models\image_encoder\config.json
### Describe the bug

I'm getting this error:

`OSError: h94/IP-Adapter does not appear to have a file named models\image_encoder\config.json. Checkout 'https://huggingface.co/h94/IP-Adapter/main' for available files.`

When running the IP-Adapter example from here: https://huggingface.co/docs/diffusers/main/en/using-diffusers/loading_adapters

### Reproduction

Run the IP-Adapter example from here: https://huggingface.co/docs/diffusers/main/en/using-diffusers/loading_adapters

```
import torch
from diffusers import StableDiffusionPipeline, DDIMScheduler
from diffusers.utils import load_image

noise_scheduler = DDIMScheduler(
    num_train_timesteps=1000,
    beta_start=0.00085,
    beta_end=0.012,
    beta_schedule="scaled_linear",
    clip_sample=False,
    set_alpha_to_one=False,
    steps_offset=1
)

pipeline = StableDiffusionPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5",
    torch_dtype=torch.float16,
    scheduler=noise_scheduler,
).to("cuda")

pipeline.load_ip_adapter("h94/IP-Adapter", subfolder="models", weight_name="ip-adapter-full-face_sd15.bin")

pipeline.set_ip_adapter_scale(0.7)

image = load_image("https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/ai_face2.png")

generator = torch.Generator(device="cpu").manual_seed(33)

image = pipeline(
    prompt="A photo of a girl wearing a black dress, holding red roses in hand, upper body, behind is the Eiffel Tower",
    ip_adapter_image=image,
    negative_prompt="monochrome, lowres, bad anatomy, worst quality, low quality", 
    num_inference_steps=50, num_images_per_prompt=1, width=512, height=704,
    generator=generator,
).images[0]
```

OSError: h94/IP-Adapter does not appear to have a file named models\image_encoder\config.json. Checkout 'https://huggingface.co/h94/IP-Adapter/main' for available files.

### Logs

```shell
OSError: h94/IP-Adapter does not appear to have a file named models\image_encoder\config.json. Checkout 'https://huggingface.co/h94/IP-Adapter/main' for available files.
```

### System Info

Win 11
Diffusers 0.25.0

### Who can help?

@sayakpaul @yiyixuxu @DN6 @patrickvonplaten

### PR DESCRIPTION
**Pull request by fabiorigano**
# What does this PR do?

Fixes #6561 <s>and https://github.com/huggingface/diffusers/pull/6192#issuecomment-1875357411</s>

@patrickvonplaten @sayakpaul

### CODE REVIEW

File: src/diffusers/loaders/ip_adapter.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def load_ip_adapter(
                 logger.info(f"loading image_encoder from {pretrained_model_name_or_path_or_dict}")
                 image_encoder = CLIPVisionModelWithProjection.from_pretrained(
                     pretrained_model_name_or_path_or_dict,
-                    subfolder=os.path.join(subfolder, "image_encoder"),
+                    subfolder=Path(subfolder, "image_encoder").as_posix(),
```
Comment (sayakpaul) [inline src/diffusers/loaders/ip_adapter.py:141]: It seems to be a good idea to follow this throughout in the library to better support Windows. 

What say @patrickvonplaten @yiyixuxu?

**Review by sayakpaul — APPROVED** [review]
Very nice PR!

File: src/diffusers/models/unet_2d_condition.py [hunk 2]
```diff
@@ -682,9 +685,11 @@ def set_default_attn_processor(self):
         Disables custom attention processors and sets the default attention implementation.
         """
         if all(proc.__class__ in ADDED_KV_ATTENTION_PROCESSORS for proc in self.attn_processors.values()):
-            processor = AttnAddedKVProcessor()
+            processor = (
+                AttnAddedKVProcessor2_0() if hasattr(F, "scaled_dot_product_attention") else AttnAddedKVProcessor()
```
[the remark this answers is not in the document]
Comment (sayakpaul) [reply src/diffusers/models/unet_2d_condition.py:689]: That should be the case indeed. No optimized things as the default.

**Comment by fabiorigano** [comment]
Removed changes on `set_default_attn_processors`!

**Review by sayakpaul — APPROVED** [review]
Super work here.

File: src/diffusers/loaders/ip_adapter.py (hunk 1 above)
Comment (sayakpaul) [reply src/diffusers/loaders/ip_adapter.py:141]: Opened a tracker here: https://github.com/huggingface/diffusers/issues/6585.

**Comment by fabiorigano** [comment]
@sayakpaul just to be sure about this fix: os.path.join seem to be working in all the other instances in the library, while it doesn't work when used in <model>.from_pretrained(os.path.join())

**Comment by sayakpaul** [comment]
That is weird. But it's good to follow a convention across the board I think.

**Pull request merged**